src/ServicePulse.Host/app/js/services/services.monitoring.js

```diff
@@ -16,6 +17,18 @@
 
             return function () {
                 clearInterval(interval);
+
+                // clear open toasts
+                for(var toastEntry in connectionToasts) {
```

Comment:
I wanted to use the notification service which seems to be the design intended for this kind of scenarios, but since it's not working I decided to go with using toastr directly. It is UI specific concern, so is the whole angular application ;) By abstracting toastr away I think you would gain the flexibility to decide where and how to render the information (e.g. a central notification error instead of toastr notifications). That seems to be a fairly easy refactoring once needed (and once the existing toast and notification services are cleaned up)